chopper/lib/src/request.dart

```diff
@@ -1,14 +1,14 @@
 import 'dart:async';
 
-import 'package:chopper/src/extensions.dart';
+import 'package:chopper/chopper.dart';
 import 'package:chopper/src/utils.dart';
 import 'package:http/http.dart' as http;
 import 'package:meta/meta.dart';
 
 /// This class represents an HTTP request that can be made with Chopper.
 class Request extends http.BaseRequest {
-  final String path;
-  final String origin;
+  final Uri uri;
```

Comment:
Hmm, why did you create another `Uri` field? Was the parent `BaseRequest.url` not good enough? I don't see it being used anywhere else besides within the class, so I don't think it's needed. Simply use the parent `BaseRequest.url`.

Reply:
Main reason was the `copyWith` of the `Request`. When creating a copy of `Request` the merged `Uri url` would be used for `uri` (or both `baseUri`) if not overridden. Having a merged `uri` with `http/https` would always go in favor of the `baseUri` and making it impossible to override the `baseUri` without also providing a `uri`.

Another solution could be removing `uri` & `baseUri` from `Request` & `copyWith` in favour of `url`. And making a factory constructor with `uri` & `baseUri` which get merged. But might introduces the query parameter copying/syncing a bit more complicated.

Reply:
I see. Tricky one, however, I'd actually prefer that `factory` constructor and clean stuff up even more, like you said.

@JEuler your thoughts?

Reply:
Hm, Let's go with what we have now, it is okay to do a bit of refactoring later. Great work, thank you for contribution @Guldem @techouse